src/buildercore/bootstrap.py

```diff
@@ -164,24 +164,42 @@ def is_resourcing():
 
     stack_all_ec2_nodes(stackname, _setup_ec2_node, username=BOOTSTRAP_USER)
 
+def remove_topics_from_sqs_policy(policy, topic_arns):
+    """Removes statements from an SQS policy.
 
-def update_sqs_stack(stackname, **kwargs):
-    pdata = project_data_for_stackname(stackname)
-    if not pdata['aws']['sqs']:
-        return
-    context = context_handler.load_context(stackname)
-    setup_sqs(stackname, context.get('sqs', {}), pdata['aws']['region'])
+    These statements are created by boto's subscribe_sqs_queue()"""
 
-def unsub_sqs(stackname, context_sqs, region, dry_run=False):
+    def for_unsubbed_topic(statement):
+        return statement.get('Condition', {}).get('StringLike', {}).get('aws:SourceArn') in topic_arns
+
+    policy['Statement'] = list(filter(lambda s: not for_unsubbed_topic(s), policy['Statement']))
```

Comment:
python2 map/filter/reduce type are not lazy, no need to realise result via `list`. 2to3 will 'helpfully' clean these up for you when builder gets upgraded. no change necessary, just sayin'.